src/GRANULAR/fix_pour.cpp

```diff
@@ -310,6 +216,108 @@ void FixPour::init()
   if (fixgrav->varflag != FixGravity::CONSTANT)
     error->all(FLERR, "Fix gravity for fix pour must be constant");
 
+  // Perform one time initialization operations
+  // outside of constructor in case fix definition precedes timestep defintion/atom creation
+
+  if (!initialize_flag) {
```

Comment:
What if an input has multiple "run" statements and the timestep is changed in between?

Reply:
I tested this and it looks like that scenario was already covered by this:
https://github.com/lammps/lammps/blob/3a53b01ed9dbd1af82e2b5af83a493b53c4760d7/src/GRANULAR/fix_pour.cpp#L1054-L1057

The only time this wasn't triggered seems to have been when the timestep was changed between the fix declaration and the first run command (so there was nothing to reset).